### PR TITLE
readme: add common causes of 500 Internal Server Error

### LINKED ISSUE #71: Issue while building docker image
I am stuck at the below error while trying to build the docker image. I was using 0.1.1 maven plugin, but even after 0.2.6 there has been no help. Can anyone please tell how to solve this?

```
Caused by: com.spotify.docker.client.shaded.javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error
        at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:968)
        at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:795)
        at org.glassfish.jersey.client.JerseyInvocation.access$500(JerseyInvocation.java:91)
        at org.glassfish.jersey.client.JerseyInvocation$5.completed(JerseyInvocation.java:756)
        at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:189)
        at org.glassfish.jersey.client.ClientRuntime.access$300(ClientRuntime.java:74)
        at org.glassfish.jersey.client.ClientRuntime$1.run(ClientRuntime.java:171)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:320)
        at org.glassfish.jersey.client.ClientRuntime$2.run(ClientRuntime.java:201)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
```

### PR DESCRIPTION
**Pull request by mattnworb**
Resolves #71.

### CODE REVIEW

**Comment by mattnworb** [comment]
@davidxia

File: README.md [hunk 1]
```diff
@@ -201,3 +203,26 @@ Exception caught: system properties: docker has type STRING rather than OBJECT
...
+Problem: when building the Docker image, Maven outputs an exception with a
+stacktrace like:
+
+> Caused by: com.spotify.docker.client.shaded.javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error
+
+docker-maven-plugin communicates with your local Docker daemon using the HTTP
+Remote API and any unexpected errors that the daemon encounters will be
+reported as `500 Internal Server Error`.
+
+Check the Docker daemon log (typically at `/var/log/docker.log`) for more details.
```
Comment (davidxia) [inline README.md]: or `/var/log/upstart/docker.log`

File: README.md [hunk 2]
```diff
@@ -201,3 +203,26 @@ Exception caught: system properties: docker has type STRING rather than OBJECT
...
+
+#### Invalid repository name ... only [a-z0-9-\_.] are allowed
+
+One common cause of `500 Internal Server Error` is attempting to build an image
+with a repository name containing uppercase characters, such as if the
+`<imageName>` in the plugin's configuration refers to `${project.version}` when
+the Maven project version is ending in `SNAPSHOT`.
+
+Consider tagging the image (with the `<dockerImageTags>` configuration option)
```
Comment (davidxia) [inline README.md]: Is the wording here a bit confusing? maybe provide an example to clarify?

**Comment by davidxia** [comment]
:+1:

**Pull request merged**